scripts/build/autotools/libbson/FindDependencies.m4

```diff
@@ -78,8 +78,12 @@ AC_CHECK_FUNC(rand_r, [AC_SUBST(BSON_HAVE_RAND_R, 1)], [], [#include <stdlib.h>]
 # Check for pthreads. We might need to make this better to handle mingw,
 # but I actually think it is okay to just check for it even though we will
 # use win32 primatives.
-AX_PTHREAD([],
-           [AC_MSG_ERROR([libbson requires pthreads on non-Windows platforms.])])
+AX_PTHREAD([
+  PHP_EVAL_INCLINE([$PTHREAD_CFLAGS])
+  PHP_EVAL_LIBLINE([$PTHREAD_LIBS],[MONGODB_SHARED_LIBADD])
```

Comment:
This is still problematic for static builds, as pthread doesn't get linked to PHP. I believe this is similar to what @derickr brought up in https://github.com/mongodb/mongo-php-driver/pull/749/files#r166189048, although I'm still curious how other extensions manage to link libraries for shared/static builds in a portable manner.

@saramg: can you chime in?